### PR TITLE
[VecOps] Fix ROOT-9771 and add example snippet to all helpers doc

### PR DESCRIPTION
**Pull request by dpiparo**

### CODE REVIEW

File: math/vecops/inc/ROOT/RVec.hxx [hunk 1]
```diff
@@ -741,6 +838,15 @@ RVec<typename RVec<T>::size_type> Argsort(const RVec<T> &v)
 }
 
 /// Return elements of a vector at given indices
+///
+/// Example code, at the ROOT prompt:
+/// ~~~{.cpp}
+/// using namespace ROOT::VecOps;
+/// RVec<double> v {2., 3., 1.};
```
Comment (stwunsch) [inline math/vecops/inc/ROOT/RVec.hxx]:
```suggestion
/// RVec<double> v {1., 2., 3.};
```

Comment (stwunsch) [reply math/vecops/inc/ROOT/RVec.hxx]: Keep it simple ;)

File: math/vecops/inc/ROOT/RVec.hxx [hunk 2]
```diff
@@ -741,6 +838,15 @@ RVec<typename RVec<T>::size_type> Argsort(const RVec<T> &v)
...
+/// auto vTaken = Take(v, {0,2});
+/// vTaken
+/// // (ROOT::VecOps::RVec<double>) { 2.0000000, 1.0000000 }
```
Comment (stwunsch) [inline math/vecops/inc/ROOT/RVec.hxx]:
```suggestion
/// // (ROOT::VecOps::RVec<double>) { 1.0000000, 3.0000000 }
```

Comment (stwunsch) [reply math/vecops/inc/ROOT/RVec.hxx]: Probably lower the precision for a nicer output in the doxygen?

File: math/vecops/inc/ROOT/RVec.hxx [hunk 3]
```diff
@@ -752,7 +858,21 @@ RVec<T> Take(const RVec<T> &v, const RVec<typename RVec<T>::size_type> &i)
...
 
-/// Return first elements of a vector if n>0 and last elements if n<0
+/// Return first or last `n` elements of an RVec
+///
+/// if `n > 0` and last elements if `n < 0`.
+///
+/// Example code, at the ROOT prompt:
+/// ~~~{.cpp}
+/// using namespace ROOT::VecOps;
+/// RVec<double> v {2., 3., 1.};
```
Comment (stwunsch) [inline math/vecops/inc/ROOT/RVec.hxx]:
```suggestion
/// RVec<double> v {1., 2., 3.};
```

File: math/vecops/inc/ROOT/RVec.hxx [hunk 4]
```diff
@@ -752,7 +858,21 @@ RVec<T> Take(const RVec<T> &v, const RVec<typename RVec<T>::size_type> &i)
...
+/// auto firstTwo = Take(v, 2);
+/// firstTwo
+/// // (ROOT::VecOps::RVec<double>) { 2.0000000, 3.0000000 }
+/// auto lastOne = Take(v, -1);
+/// lastOne
+/// // (ROOT::VecOps::RVec<double>) { 1.0000000 }
```
Comment (stwunsch) [inline math/vecops/inc/ROOT/RVec.hxx]:
```suggestion
/// // (ROOT::VecOps::RVec<double>) { 3.0000000 }
```

File: math/vecops/inc/ROOT/RVec.hxx [hunk 5]
```diff
@@ -752,7 +858,21 @@ RVec<T> Take(const RVec<T> &v, const RVec<typename RVec<T>::size_type> &i)
...
+///
+/// if `n > 0` and last elements if `n < 0`.
+///
+/// Example code, at the ROOT prompt:
+/// ~~~{.cpp}
+/// using namespace ROOT::VecOps;
+/// RVec<double> v {2., 3., 1.};
+/// auto firstTwo = Take(v, 2);
+/// firstTwo
+/// // (ROOT::VecOps::RVec<double>) { 2.0000000, 3.0000000 }
```
Comment (stwunsch) [inline math/vecops/inc/ROOT/RVec.hxx]:
```suggestion
/// // (ROOT::VecOps::RVec<double>) { 1.0000000, 2.0000000 }
```

File: math/vecops/inc/ROOT/RVec.hxx [hunk 6]
```diff
@@ -776,6 +896,15 @@ RVec<T> Take(const RVec<T> &v, const int n)
 }
 
 /// Return copy of reversed vector
+///
+/// Example code, at the ROOT prompt:
+/// ~~~{.cpp}
+/// using namespace ROOT::VecOps;
+/// RVec<double> v {2., 3., 1.};
```
Comment (stwunsch) [inline math/vecops/inc/ROOT/RVec.hxx]:
```suggestion
/// RVec<double> v {1., 2., 3.};
```

File: math/vecops/inc/ROOT/RVec.hxx [hunk 7]
```diff
@@ -776,6 +896,15 @@ RVec<T> Take(const RVec<T> &v, const int n)
...
+/// auto v_reverse = Reverse(v);
+/// v_reverse
+/// // (ROOT::VecOps::RVec<double>) { 1.0000000, 3.0000000, 2.0000000 }
```
Comment (stwunsch) [inline math/vecops/inc/ROOT/RVec.hxx]:
```suggestion
/// // (ROOT::VecOps::RVec<double>) { 3.0000000, 2.0000000, 1.0000000 }
```

File: math/vecops/inc/ROOT/RVec.hxx [hunk 8]
```diff
@@ -906,6 +1110,17 @@ RVec<T> Intersect(const RVec<T>& v1, const RVec<T>& v2, bool v2_is_sorted = fals
...
 /// Return the elements of v1 if the condition c is true and v2 if the
 /// condition c is false.
+///
+/// Example code, at the ROOT prompt:
+/// ~~~{.cpp}
+/// using namespace ROOT::VecOps;
+/// RVec<double> v1 {1., 2., 3.};
+/// RVec<double> v2 {-1., -2., -3.};
+/// RVec<int> c {0, 1, 1};
+/// auto if_c_v1_else_v2 = Where(c, v1, v2);
```
Comment (stwunsch) [inline math/vecops/inc/ROOT/RVec.hxx]:
```suggestion
/// auto if_c_v1_else_v2 = Where(v1>1, v1, v2);
```

Comment (stwunsch) [reply math/vecops/inc/ROOT/RVec.hxx]: `RVec<int>` will nobody write explicitly!

File: math/vecops/inc/ROOT/RVec.hxx [hunk 9]
```diff
@@ -921,6 +1136,17 @@ RVec<T> Where(const RVec<int>& c, const RVec<T>& v1, const RVec<T>& v2)
...
 /// Return the elements of v1 if the condition c is true and sets the value v2
 /// if the condition c is false.
+///
+/// Example code, at the ROOT prompt:
+/// ~~~{.cpp}
+/// using namespace ROOT::VecOps;
+/// RVec<double> v1 {1., 2., 3.};
+/// double v2 = 4.;
+/// RVec<int> c {0, 1, 1};
+/// auto if_c_v1_else_v2 = Where(c, v1, v2);
```
Comment (stwunsch) [inline math/vecops/inc/ROOT/RVec.hxx]:
```suggestion
/// auto if_c_v1_else_v2 = Where(v1>1, v1, v2);
```

File: math/vecops/inc/ROOT/RVec.hxx [hunk 10]
```diff
@@ -936,6 +1162,17 @@ RVec<T> Where(const RVec<int>& c, const RVec<T>& v1, T v2)
...
 /// Return the elements of v2 if the condition c is false and sets the value v1
 /// if the condition c is true.
+///
+/// Example code, at the ROOT prompt:
+/// ~~~{.cpp}
+/// using namespace ROOT::VecOps;
+/// double v1 = 4.;
+/// RVec<double> v2 {1., 2., 3.};
+/// RVec<int> c {0, 1, 1};
+/// auto if_c_v1_else_v2 = Where(c, v1, v2);
```
Comment (stwunsch) [inline math/vecops/inc/ROOT/RVec.hxx]:
```suggestion
/// auto if_c_v1_else_v2 = Where(v1>1, v1, v2);
```

File: math/vecops/inc/ROOT/RVec.hxx (hunk 8 above)
Comment (stwunsch) [reply math/vecops/inc/ROOT/RVec.hxx]: In case you agree, don't forget to remove the condition `c`.

File: math/vecops/inc/ROOT/RVec.hxx [hunk 11]
```diff
@@ -951,6 +1188,17 @@ RVec<T> Where(const RVec<int>& c, T v1, const RVec<T>& v2)
...
 /// Return a vector with the value v2 if the condition c is false and sets the
 /// value v1 if the condition c is true.
+///
+/// Example code, at the ROOT prompt:
+/// ~~~{.cpp}
+/// using namespace ROOT::VecOps;
+/// double v1 = 4.;
+/// double v2 = 2.;
+/// RVec<int> c {0, 1, 1};
+/// auto if_c_v1_else_v2 = Where(c, v1, v2);
```
Comment (stwunsch) [inline math/vecops/inc/ROOT/RVec.hxx]:
```suggestion
/// auto if_c_v1_else_v2 = Where(v1>1, v1, v2);
```

File: math/vecops/inc/ROOT/RVec.hxx (hunk 2 above)
Comment (dpiparo) [reply math/vecops/inc/ROOT/RVec.hxx]: I would not. People often complain about what they get at the prompt and what we say they'd get. So, probably lower the precision in the value printer? ;-)

File: math/vecops/inc/ROOT/RVec.hxx (hunk 8 above)
Comment (dpiparo) [reply math/vecops/inc/ROOT/RVec.hxx]: Here perhaps one need to be more explicit. I would show what c is in any case.

**Pull request merged**